src/test/java/plugins/LogParserTest.java

```diff
@@ -63,6 +62,58 @@ public void trendVisible(){
     }
 
     /**
+     * Test case:
+     * Check for the build to be marked as unstable.
+     */
+    @Test
+    public void checkMarkedUnstableOnWarning() {
+        // Create a special rule set for this test case
+        Resource rule = resource("/logparser_plugin/rules/log-parser-rule-markings");
+        addLogParserRule("error_warning_rule", rule.url.getPath());
+
+        // Create a new freestyle job
+        FreeStyleJob job = jenkins.jobs.create(FreeStyleJob.class, "simple-job");
+        job.configure(() -> {
+            job.addShellStep("echo marked as warn");
+            LogParserPublisher lpp = job.addPublisher(LogParserPublisher.class);
+            lpp.setMarkOnUnstableWarning(true);
+            lpp.setRule(LogParserPublisher.RuleType.GLOBAL, "" + rule.url.getPath());
+        });
+
+        job.startBuild().waitUntilFinished().open();
+
+        // Find the status image and check if it is set correctly
+        WebElement imgUnstable = driver.findElement(By.xpath("//div[@id='main-panel']/h1[@class='build-caption page-headline']/img"));
```

Comment:
Wouldn't it bee simpler to call `job.startBuild().waitUntilFinished().shouldBeUnstable()`